diff --git a/dist/assets/services/service.js b/dist/assets/services/service.js
index 42e9dc464f6b7275174b8186670339965cbc3889..c66841846605908fa092cac387ecbdfaf92b8693 100644
--- a/dist/assets/services/service.js
+++ b/dist/assets/services/service.js
@@ -55,13 +55,14 @@ const baseService = {
           )
         });
       }
-      if (options.src.format === "svg") {
-        options.format = "svg";
+      if (options.src.format === 'svg' || options.src.format === 'gif') {
+        options.format = options.src.format
       }
     }
     if (!options.format) {
       options.format = "webp";
     }
+    options.process = !(options.src.format === options.format)
     return options;
   },
   getHTMLAttributes(options) {
diff --git a/dist/assets/services/squoosh.js b/dist/assets/services/squoosh.js
index a22d039d2428dcb1fd9ee54685ff82c275cf5eb1..bf68e6cad508e1bd03717b0b9bbf07f64c33057d 100644
--- a/dist/assets/services/squoosh.js
+++ b/dist/assets/services/squoosh.js
@@ -44,8 +44,9 @@ const service = {
   async transform(inputBuffer, transformOptions) {
     const transform = transformOptions;
     let format = transform.format;
-    if (format === "svg")
-      return { data: inputBuffer, format: "svg" };
+    if (!transform.process) {
+      return { data: inputBuffer, format }
+    }
     const operations = [];
     const rotation = await getRotationForEXIF(transform, inputBuffer);
     if (rotation) {
diff --git a/dist/vite-plugin-markdown/index.js b/dist/vite-plugin-markdown/index.js
index 4beaee1174892d2ece9f6ac5059d428d93858514..5b044f85f36e822e8887b176f96891faaf0b8113 100644
--- a/dist/vite-plugin-markdown/index.js
+++ b/dist/vite-plugin-markdown/index.js
@@ -118,7 +118,7 @@ function markdown({ settings, logging }) {
 
 				function updateImageReferences(html) {
 					return html.replaceAll(
-						/__ASTRO_IMAGE_="(.+)"/gm,
+						/__ASTRO_IMAGE_="(.+?)"/gm,
 						(full, imagePath) => spreadAttributes({src: images[imagePath].src, ...images[imagePath].attributes})
 					);
 				}